src/VirtualClient/VirtualClient.Actions.UnitTests/StressAppTest/StressAppTestMetricsParserTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using NUnit.Framework;
+using VirtualClient.Contracts;
+using VirtualClient.Common.Contracts;
+using VirtualClient.Actions;
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Actions
+{
+    [TestFixture]
+    [Category("Unit")]
+    class StressAppTestMetricsParserTests
+    {
+        private StressAppTestMetricsParser testParser;
+        string workingDirectory;
+
+        [SetUp]
+        public void Setup()
+        {
+            this.workingDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+        }
+
+        [Test]
+        public void StressAppTestParserVerifyMetricsForPassResults()
+        {
+            string resultsPath = Path.Combine(this.workingDirectory, "Examples", "StressAppTest", "stressAppTestLog_pass.txt");
+            string rawText = File.ReadAllText(resultsPath);
+            this.testParser = new StressAppTestMetricsParser(rawText);
+            IList<Metric> metrics = this.testParser.Parse();
+
+            Assert.AreEqual(1, metrics.Count);
+            MetricAssert.Exists(metrics, "hardwareErrorCount", 0);
```

Comment:
Do we not parse the performance, there is a memory performance metric?

Reply:
StressAppTest is a stability stress workload, that stresses out the Memory interface with randomized traffic to memory from processor and I/O. The intent is to create a realistic high load situation and see if there is any hardware crash/error in that scenario to Qualify systems.
The metricsParser has been modified to capture exact Hardware errors, as reported by StressAppTest, in the form of Tags of the metric for ease in debug.